treeherder/etl/perf_data_adapters.py

```diff
@@ -142,6 +160,75 @@ def __init__(self):
         self.performance_artifact_placeholders = []
         self.signature_property_placeholders = []
 
+
+    @staticmethod
+    def _get_base_perf_obj(job_guid, name, type, talos_datum, series_signature,
+                           signature_properties, series_data):
+
+        # N.B. test_build information (i.e. revision) just there to ease third
+        # party alert processing, since we should normally be able to get
+        # it by querying the job info
+        obj = {
+            "job_guid": job_guid,
+            "name": name,
+            "type": type,
+            "blob": {
+                "date": talos_datum["testrun"]["date"],
+                "series_signature": series_signature,
+                "signature_properties": signature_properties,
+                "performance_series": series_data,
+                "testsuite": talos_datum["testrun"]["suite"],
+                "metadata":{ 'test_build': talos_datum['test_build'] }
+            }
+        }
+        options = talos_datum["testrun"].get("options")
+        if options:
+            obj['blob']['metadata']['options'] = options
+
+        return obj
+
+
+    @staticmethod
+    def _get_series_signature(signature_properties):
+        signature_prop_values = signature_properties.keys()
+        signature_prop_values.extend(signature_properties.values())
+
+        sha = sha1()
+        sha.update(''.join(map(lambda x: str(x), sorted(signature_prop_values))))
+
+        return sha.hexdigest()
+
+
+    def _update_signatures(self, series_signature, signature_properties):
+        if series_signature not in self.signatures:
+            self.signatures[series_signature] = []
+
```

Comment:
So this commit was slightly broken, I have an updated one where this code is in _add_performance_artifact